packages/components/src/navigator/navigator-back-link/README.md

```diff
@@ -0,0 +1,27 @@
+# `NavigatorBackLink`
```

Comment:
Should we care that these components are semantically buttons rather than links? I generally wouldn't name this a `Link` unless it started modifying `window.location`, but I wonder where your thoughts are.

Reply:
I personally liked the name "link" because it's commonly associated to the element/component that enables navigation to a different page/screen, but I also agree with your point.

I went ahead and renamed `NavigatorLink` to `NavigatorButton` and `NavigatorBackLink` to `NavigatorBackButton` in https://github.com/WordPress/gutenberg/pull/38634/commits/7f1b2572015f6e7d25833b0f32e046524d80c94e, let me know if you had another name in mind instead.